cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.tools;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.queue.QueueName;
+import co.cask.cdap.data2.transaction.queue.QueueConstants;
+import co.cask.cdap.data2.transaction.queue.QueueEntryRow;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.proto.Id;
+import com.google.common.base.Joiner;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.NavigableMap;
+import javax.annotation.Nullable;
+
+/**
+ * Upgrades queue.config table
+ */
+public class QueueConfigUpgrader extends AbstractUpgrader {
+  private static final Logger LOG = LoggerFactory.getLogger(QueueConfigUpgrader.class);
+  private final HBaseTableUtil tableUtil;
+
+  @Inject
+  public QueueConfigUpgrader(LocationFactory locationFactory, HBaseTableUtil tableUtil) {
+    super(locationFactory);
+    this.tableUtil = tableUtil;
+  }
+
+  @Override
+  void upgrade() throws Exception {
+    String dsName = Joiner.on(".").join(Constants.SYSTEM_NAMESPACE, QueueConstants.QUEUE_CONFIG_TABLE_NAME);
+    Id.DatasetInstance datasetId = Id.DatasetInstance.from(Constants.DEFAULT_NAMESPACE_ID, dsName);
+    HTable queueConfigTable = tableUtil.createHTable(HBaseConfiguration.create(),
+                                                     TableId.from(datasetId.getNamespaceId(), dsName));
+    LOG.info("Starting upgrade for queue config table {}", Bytes.toString(queueConfigTable.getTableName()));
+    try {
+      Scan scan = new Scan();
+      scan.setTimeRange(0, HConstants.LATEST_TIMESTAMP);
+      scan.addFamily(QueueEntryRow.COLUMN_FAMILY);
+      scan.setMaxVersions(1); // we only need to see one version of each row
+      ResultScanner resultScanner = queueConfigTable.getScanner(scan);
```

Comment:
As a convention, the try-finally should start right after the resource you need to free-up (meaning the `try` should be here, not after new array list).